cogs/onmyoji.py

```diff
@@ -1,18 +1,110 @@
+from __future__ import print_function
 import discord
 import json
 from discord.ext import commands
+import os.path
+from googleapiclient.discovery import build
+from googleapiclient.http import *
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+from google.oauth2 import service_account
+import io
+import csv
+import config
+import re
 
+permission = 'You do not have permission to use this command.'
+owner_list = config.owner_list
+editor_list = config.editor_list
 
-with open('lists/bounty.json') as file:
-    bounty_json = json.loads(file.read())
+recommend = re.compile(r'recommend',re.IGNORECASE)
 
 with open('lists/stats.json') as file:
     stats_json = json.loads(file.read())
 
+def bold(input):
+    '''Returns the Discord bolded version of input text.'''
+    return "**"+input+"**"
+
+class DriveAPI:
+    '''Manages the Google Drive API Auth and retreiving the CSV database.'''
+    SCOPES = ['https://www.googleapis.com/auth/drive.readonly']
+    SERVICE_ACCOUNT_FILE = 'bbt_credentials.json'
+    credentials = service_account.Credentials.from_service_account_file(
+            SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+    drive_service = build('drive', 'v3', credentials = credentials)
+
+    @classmethod
+    def get_gdrive_sheet_database(cls):
+        file_id = config.bounty_file_id
+        request = cls.drive_service.files().export_media(fileId=file_id, mimeType='text/csv')
+        buffer_file = io.BytesIO()
+        downloader = MediaIoBaseDownload(buffer_file, request)
+        done = False
+        while done is False:
+            status, done = downloader.next_chunk()
+            print ("Download %d%%." % int(status.progress() * 100))
+        with open ('lists/shiki_bounty.csv', 'wb') as f:
+            f.write(buffer_file.getvalue())
+        return
+
+
+class Shikigami:
+    def __init__(self, input_name, bounty_db):
+        self.name = input_name
+        for row in bounty_db:
+            shiki_name, alias, hints, locations = row[0], row[1], row[2], row[3]
+            if input_name in shiki_name:
+                self.alias = alias
+                self.hints = hints
+                self.locations = locations.split('\n')
+            elif self.name.lower() in alias.lower():
+                self.name = shiki_name
+                self.alias = alias
+                self.hints = hints
+                self.locations = locations
 
 class Onmyoji(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
+        try:
+            self.create_classes()
+        except FileNotFoundError:
+            DriveAPI.class_get_gdrive_sheet_database()
+            self.create_classes()
+
+    async def has_permission(ctx):
+        if ctx.author.id in owner_list or ctx.author.id in editor_list:
```

Comment:
Bad pattern, if you find yourself returning a bool you probably just wanna do
```suggestion
        return ctx.author.id in owner_list or ctx.author.id in editor_list:
```